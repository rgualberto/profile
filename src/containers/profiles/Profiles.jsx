import './profiles.scss';
import React, {Component} from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import {push} from 'react-router-redux';
import Page from '../page/Page.jsx';
import ProfileCard from '../../components/profile/ProfileCard.jsx';

export class Profiles extends Component {
  handleRoute(event) {
    event.preventDefault();

    const nextRoute = event.target.attributes.href.value;

    this.props.push(nextRoute);
  }

  render() {
    const {profiles} = this.props;

    return (
      <Page>
        <div className="profiles">
          {
            profiles.map((profile, index) => (
              <ProfileCard
                key={index}
                profile={profile}
                handleRoute={this.handleRoute.bind(this)}
              />
            ))
          }
        </div>
      </Page>
    );
  }
}

Profiles.propTypes = {
  profiles: PropTypes.array.isRequired,
  push: PropTypes.func.isRequired,
};

function mapStateToProps(state) {
  return {
    profiles: state.profilesReducer.profiles
  };
}

function mapDispatchToProps(dispatch) {
  return {
  push: bindActionCreators(push, dispatch)};
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Profiles);
