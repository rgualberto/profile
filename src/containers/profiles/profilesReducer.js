import _ from 'lodash';
export const HYDRATE_PROFILES = 'profile/HYDRATE_PROFILES';
export const UPDATE_PROFILE = 'profile/UPDATE_PROFILE';

export const initialState = {
  profiles: [
    {
      userId: 123,
      photo: "http://lorempixel.com/160/160/",
      name: "Sven Motorson",
      location: "Boston",
      education: [
        {
          degree: "Bachelors of Engineering: Mechanical",
          school: "Carnegie Mellon University"
        },
        {
          degree: "Masters of Engineering: Mechanical",
          school: "Carnegie Mellon University"
        },
        {
          degree: "Docterates of Engineering: Mechanical",
          school: "Massachusetts Institute of Technology"
        }
      ],
      currentProject: "Duis in nunc ultrices, viverra lorem sed, suscipit mauris. Nullam maximus, sapien non facilisis mattis, magna ex sodales ipsum, eget tristique sapien ante quis libero. Praesent pretium velit ac odio egestas fringilla. Cras nec tortor dignissim, efficitur ligula nec, imperdiet odio. Etiam in augue felis. Duis iaculis placerat vulputate. Aliquam in mauris tempor risus egestas blandit ac ut lacus.",
      wikis: [
        {
          title: "Engine",
          uriParam: "engine"
        },
        {
          title: "Electric motor",
          uriParam: "Electric_motor"
        }
      ]
    },
    {
      userId: 456,
      photo: "http://lorempixel.com/160/160/",
      name: "Alice Mendelson",
      location: "Baltimore",
      education: [
        {
          degree: "Bachelors of Science: Biomedical Science",
          school: "Rutgers University"
        },
        {
          degree: "Masters of Science: Biomedical Science",
          school: "Rutgers University"
        },
        {
          degree: "Docterates of Science: Biomedical Science",
          school: "Johns Hopkins University"
        }
      ],
      currentProject: "In mattis arcu arcu, convallis volutpat eros consequat vitae. Pellentesque ultricies vulputate magna nec facilisis. Nullam consectetur risus orci, tincidunt molestie libero tempus eu. Vivamus a mauris vel nunc gravida laoreet finibus eu erat. Vestibulum sollicitudin volutpat nisl ac rhoncus. Nulla volutpat elit nec ante molestie, lacinia egestas neque molestie. Nulla nulla purus, ornare non vehicula at, blandit vel purus. Vivamus vitae ipsum id dui porttitor malesuada aliquam quis lacus. Cras quam nisi, finibus ac tellus quis, lobortis convallis nisl. Mauris viverra imperdiet placerat. Ut in eros sit amet ligula viverra luctus. Etiam ultrices ultricies pellentesque. Fusce malesuada vehicula egestas. Orci varius natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus.",
      wikis: []
    }
  ]
};

export const profilesReducer = (state = initialState, action = {}) => {
    switch (action.type) {
      case HYDRATE_PROFILES: {
        const existingData = window.localStorage.getItem('profileData');
        const profiles = _.isNull(existingData) ? state.profiles : JSON.parse(existingData);

        return {
          ...state,
          profiles
        };
      }
      case UPDATE_PROFILE: {
        const newProfiles = [
          ..._.reject(state.profiles, ['userId', action.userId]),
          {
            ..._.find(state.profiles, ['userId', action.userId]),
            ...action.profileData
          }
        ];

        window.localStorage.setItem('profileData', JSON.stringify(newProfiles));

        return {
          ...state,
          profiles: newProfiles
        };
      }
      default:
        return state;
    }
};

export const hydrateProfiles = () => ({ type: HYDRATE_PROFILES });

export const updateProfile = (userId, profileData) => ({
  type: UPDATE_PROFILE,
  userId,
  profileData
});

export default profilesReducer;
