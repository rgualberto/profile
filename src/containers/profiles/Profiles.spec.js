import React from 'react';
import {Profiles} from './Profiles.jsx';
import {shallow} from 'enzyme';
import {initialState} from './profilesReducer';

describe('Profiles Container', () => {
  it('renders profiles', () => {
    const wrapper = shallow(
      <Profiles
        profiles={initialState.profiles}
        push={() => undefined}
      />);

    expect(wrapper).toMatchSnapshot();
  });
});
