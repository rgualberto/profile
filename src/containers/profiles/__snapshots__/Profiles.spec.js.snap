// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profiles Container renders profiles 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Connect(Page)>
    <div
        className="profiles"
    >
        <ProfileCard
            handleRoute={[Function]}
            profile={
                Object {
                    "currentProject": "Duis in nunc ultrices, viverra lorem sed, suscipit mauris. Nullam maximus, sapien non facilisis mattis, magna ex sodales ipsum, eget tristique sapien ante quis libero. Praesent pretium velit ac odio egestas fringilla. Cras nec tortor dignissim, efficitur ligula nec, imperdiet odio. Etiam in augue felis. Duis iaculis placerat vulputate. Aliquam in mauris tempor risus egestas blandit ac ut lacus.",
                    "education": Array [
                      Object {
                        "degree": "Bachelors of Engineering: Mechanical",
                        "school": "Carnegie Mellon University",
                      },
                      Object {
                        "degree": "Masters of Engineering: Mechanical",
                        "school": "Carnegie Mellon University",
                      },
                      Object {
                        "degree": "Docterates of Engineering: Mechanical",
                        "school": "Massachusetts Institute of Technology",
                      },
                    ],
                    "location": "Boston",
                    "name": "Sven Motorson",
                    "photo": "http://lorempixel.com/160/160/",
                    "userId": 123,
                    "wikis": Array [
                      Object {
                        "title": "Engine",
                        "uriParam": "engine",
                      },
                      Object {
                        "title": "Electric motor",
                        "uriParam": "Electric_motor",
                      },
                    ],
                  }
            }
        />
        <ProfileCard
            handleRoute={[Function]}
            profile={
                Object {
                    "currentProject": "In mattis arcu arcu, convallis volutpat eros consequat vitae. Pellentesque ultricies vulputate magna nec facilisis. Nullam consectetur risus orci, tincidunt molestie libero tempus eu. Vivamus a mauris vel nunc gravida laoreet finibus eu erat. Vestibulum sollicitudin volutpat nisl ac rhoncus. Nulla volutpat elit nec ante molestie, lacinia egestas neque molestie. Nulla nulla purus, ornare non vehicula at, blandit vel purus. Vivamus vitae ipsum id dui porttitor malesuada aliquam quis lacus. Cras quam nisi, finibus ac tellus quis, lobortis convallis nisl. Mauris viverra imperdiet placerat. Ut in eros sit amet ligula viverra luctus. Etiam ultrices ultricies pellentesque. Fusce malesuada vehicula egestas. Orci varius natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus.",
                    "education": Array [
                      Object {
                        "degree": "Bachelors of Science: Biomedical Science",
                        "school": "Rutgers University",
                      },
                      Object {
                        "degree": "Masters of Science: Biomedical Science",
                        "school": "Rutgers University",
                      },
                      Object {
                        "degree": "Docterates of Science: Biomedical Science",
                        "school": "Johns Hopkins University",
                      },
                    ],
                    "location": "Baltimore",
                    "name": "Alice Mendelson",
                    "photo": "http://lorempixel.com/160/160/",
                    "userId": 456,
                    "wikis": Array [],
                  }
            }
        />
    </div>
</Connect(Page)>,
  "nodes": Array [
    <Connect(Page)>
      <div
            className="profiles"
      >
            <ProfileCard
                  handleRoute={[Function]}
                  profile={
                        Object {
                              "currentProject": "Duis in nunc ultrices, viverra lorem sed, suscipit mauris. Nullam maximus, sapien non facilisis mattis, magna ex sodales ipsum, eget tristique sapien ante quis libero. Praesent pretium velit ac odio egestas fringilla. Cras nec tortor dignissim, efficitur ligula nec, imperdiet odio. Etiam in augue felis. Duis iaculis placerat vulputate. Aliquam in mauris tempor risus egestas blandit ac ut lacus.",
                              "education": Array [
                                Object {
                                  "degree": "Bachelors of Engineering: Mechanical",
                                  "school": "Carnegie Mellon University",
                                },
                                Object {
                                  "degree": "Masters of Engineering: Mechanical",
                                  "school": "Carnegie Mellon University",
                                },
                                Object {
                                  "degree": "Docterates of Engineering: Mechanical",
                                  "school": "Massachusetts Institute of Technology",
                                },
                              ],
                              "location": "Boston",
                              "name": "Sven Motorson",
                              "photo": "http://lorempixel.com/160/160/",
                              "userId": 123,
                              "wikis": Array [
                                Object {
                                  "title": "Engine",
                                  "uriParam": "engine",
                                },
                                Object {
                                  "title": "Electric motor",
                                  "uriParam": "Electric_motor",
                                },
                              ],
                            }
                  }
            />
            <ProfileCard
                  handleRoute={[Function]}
                  profile={
                        Object {
                              "currentProject": "In mattis arcu arcu, convallis volutpat eros consequat vitae. Pellentesque ultricies vulputate magna nec facilisis. Nullam consectetur risus orci, tincidunt molestie libero tempus eu. Vivamus a mauris vel nunc gravida laoreet finibus eu erat. Vestibulum sollicitudin volutpat nisl ac rhoncus. Nulla volutpat elit nec ante molestie, lacinia egestas neque molestie. Nulla nulla purus, ornare non vehicula at, blandit vel purus. Vivamus vitae ipsum id dui porttitor malesuada aliquam quis lacus. Cras quam nisi, finibus ac tellus quis, lobortis convallis nisl. Mauris viverra imperdiet placerat. Ut in eros sit amet ligula viverra luctus. Etiam ultrices ultricies pellentesque. Fusce malesuada vehicula egestas. Orci varius natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus.",
                              "education": Array [
                                Object {
                                  "degree": "Bachelors of Science: Biomedical Science",
                                  "school": "Rutgers University",
                                },
                                Object {
                                  "degree": "Masters of Science: Biomedical Science",
                                  "school": "Rutgers University",
                                },
                                Object {
                                  "degree": "Docterates of Science: Biomedical Science",
                                  "school": "Johns Hopkins University",
                                },
                              ],
                              "location": "Baltimore",
                              "name": "Alice Mendelson",
                              "photo": "http://lorempixel.com/160/160/",
                              "userId": 456,
                              "wikis": Array [],
                            }
                  }
            />
      </div>
</Connect(Page)>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Profiles
        profiles={
                Array [
                        Object {
                          "currentProject": "Duis in nunc ultrices, viverra lorem sed, suscipit mauris. Nullam maximus, sapien non facilisis mattis, magna ex sodales ipsum, eget tristique sapien ante quis libero. Praesent pretium velit ac odio egestas fringilla. Cras nec tortor dignissim, efficitur ligula nec, imperdiet odio. Etiam in augue felis. Duis iaculis placerat vulputate. Aliquam in mauris tempor risus egestas blandit ac ut lacus.",
                          "education": Array [
                            Object {
                              "degree": "Bachelors of Engineering: Mechanical",
                              "school": "Carnegie Mellon University",
                            },
                            Object {
                              "degree": "Masters of Engineering: Mechanical",
                              "school": "Carnegie Mellon University",
                            },
                            Object {
                              "degree": "Docterates of Engineering: Mechanical",
                              "school": "Massachusetts Institute of Technology",
                            },
                          ],
                          "location": "Boston",
                          "name": "Sven Motorson",
                          "photo": "http://lorempixel.com/160/160/",
                          "userId": 123,
                          "wikis": Array [
                            Object {
                              "title": "Engine",
                              "uriParam": "engine",
                            },
                            Object {
                              "title": "Electric motor",
                              "uriParam": "Electric_motor",
                            },
                          ],
                        },
                        Object {
                          "currentProject": "In mattis arcu arcu, convallis volutpat eros consequat vitae. Pellentesque ultricies vulputate magna nec facilisis. Nullam consectetur risus orci, tincidunt molestie libero tempus eu. Vivamus a mauris vel nunc gravida laoreet finibus eu erat. Vestibulum sollicitudin volutpat nisl ac rhoncus. Nulla volutpat elit nec ante molestie, lacinia egestas neque molestie. Nulla nulla purus, ornare non vehicula at, blandit vel purus. Vivamus vitae ipsum id dui porttitor malesuada aliquam quis lacus. Cras quam nisi, finibus ac tellus quis, lobortis convallis nisl. Mauris viverra imperdiet placerat. Ut in eros sit amet ligula viverra luctus. Etiam ultrices ultricies pellentesque. Fusce malesuada vehicula egestas. Orci varius natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus.",
                          "education": Array [
                            Object {
                              "degree": "Bachelors of Science: Biomedical Science",
                              "school": "Rutgers University",
                            },
                            Object {
                              "degree": "Masters of Science: Biomedical Science",
                              "school": "Rutgers University",
                            },
                            Object {
                              "degree": "Docterates of Science: Biomedical Science",
                              "school": "Johns Hopkins University",
                            },
                          ],
                          "location": "Baltimore",
                          "name": "Alice Mendelson",
                          "photo": "http://lorempixel.com/160/160/",
                          "userId": 456,
                          "wikis": Array [],
                        },
                      ]
        }
        push={[Function]}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Profiles {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "profiles": Array [
            Object {
              "currentProject": "Duis in nunc ultrices, viverra lorem sed, suscipit mauris. Nullam maximus, sapien non facilisis mattis, magna ex sodales ipsum, eget tristique sapien ante quis libero. Praesent pretium velit ac odio egestas fringilla. Cras nec tortor dignissim, efficitur ligula nec, imperdiet odio. Etiam in augue felis. Duis iaculis placerat vulputate. Aliquam in mauris tempor risus egestas blandit ac ut lacus.",
              "education": Array [
                Object {
                  "degree": "Bachelors of Engineering: Mechanical",
                  "school": "Carnegie Mellon University",
                },
                Object {
                  "degree": "Masters of Engineering: Mechanical",
                  "school": "Carnegie Mellon University",
                },
                Object {
                  "degree": "Docterates of Engineering: Mechanical",
                  "school": "Massachusetts Institute of Technology",
                },
              ],
              "location": "Boston",
              "name": "Sven Motorson",
              "photo": "http://lorempixel.com/160/160/",
              "userId": 123,
              "wikis": Array [
                Object {
                  "title": "Engine",
                  "uriParam": "engine",
                },
                Object {
                  "title": "Electric motor",
                  "uriParam": "Electric_motor",
                },
              ],
            },
            Object {
              "currentProject": "In mattis arcu arcu, convallis volutpat eros consequat vitae. Pellentesque ultricies vulputate magna nec facilisis. Nullam consectetur risus orci, tincidunt molestie libero tempus eu. Vivamus a mauris vel nunc gravida laoreet finibus eu erat. Vestibulum sollicitudin volutpat nisl ac rhoncus. Nulla volutpat elit nec ante molestie, lacinia egestas neque molestie. Nulla nulla purus, ornare non vehicula at, blandit vel purus. Vivamus vitae ipsum id dui porttitor malesuada aliquam quis lacus. Cras quam nisi, finibus ac tellus quis, lobortis convallis nisl. Mauris viverra imperdiet placerat. Ut in eros sit amet ligula viverra luctus. Etiam ultrices ultricies pellentesque. Fusce malesuada vehicula egestas. Orci varius natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus.",
              "education": Array [
                Object {
                  "degree": "Bachelors of Science: Biomedical Science",
                  "school": "Rutgers University",
                },
                Object {
                  "degree": "Masters of Science: Biomedical Science",
                  "school": "Rutgers University",
                },
                Object {
                  "degree": "Docterates of Science: Biomedical Science",
                  "school": "Johns Hopkins University",
                },
              ],
              "location": "Baltimore",
              "name": "Alice Mendelson",
              "photo": "http://lorempixel.com/160/160/",
              "userId": 456,
              "wikis": Array [],
            },
          ],
          "push": [Function],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Connect(Page)>
          <div
                    className="profiles"
          >
                    <ProfileCard
                              handleRoute={[Function]}
                              profile={
                                        Object {
                                                  "currentProject": "Duis in nunc ultrices, viverra lorem sed, suscipit mauris. Nullam maximus, sapien non facilisis mattis, magna ex sodales ipsum, eget tristique sapien ante quis libero. Praesent pretium velit ac odio egestas fringilla. Cras nec tortor dignissim, efficitur ligula nec, imperdiet odio. Etiam in augue felis. Duis iaculis placerat vulputate. Aliquam in mauris tempor risus egestas blandit ac ut lacus.",
                                                  "education": Array [
                                                    Object {
                                                      "degree": "Bachelors of Engineering: Mechanical",
                                                      "school": "Carnegie Mellon University",
                                                    },
                                                    Object {
                                                      "degree": "Masters of Engineering: Mechanical",
                                                      "school": "Carnegie Mellon University",
                                                    },
                                                    Object {
                                                      "degree": "Docterates of Engineering: Mechanical",
                                                      "school": "Massachusetts Institute of Technology",
                                                    },
                                                  ],
                                                  "location": "Boston",
                                                  "name": "Sven Motorson",
                                                  "photo": "http://lorempixel.com/160/160/",
                                                  "userId": 123,
                                                  "wikis": Array [
                                                    Object {
                                                      "title": "Engine",
                                                      "uriParam": "engine",
                                                    },
                                                    Object {
                                                      "title": "Electric motor",
                                                      "uriParam": "Electric_motor",
                                                    },
                                                  ],
                                                }
                              }
                    />
                    <ProfileCard
                              handleRoute={[Function]}
                              profile={
                                        Object {
                                                  "currentProject": "In mattis arcu arcu, convallis volutpat eros consequat vitae. Pellentesque ultricies vulputate magna nec facilisis. Nullam consectetur risus orci, tincidunt molestie libero tempus eu. Vivamus a mauris vel nunc gravida laoreet finibus eu erat. Vestibulum sollicitudin volutpat nisl ac rhoncus. Nulla volutpat elit nec ante molestie, lacinia egestas neque molestie. Nulla nulla purus, ornare non vehicula at, blandit vel purus. Vivamus vitae ipsum id dui porttitor malesuada aliquam quis lacus. Cras quam nisi, finibus ac tellus quis, lobortis convallis nisl. Mauris viverra imperdiet placerat. Ut in eros sit amet ligula viverra luctus. Etiam ultrices ultricies pellentesque. Fusce malesuada vehicula egestas. Orci varius natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus.",
                                                  "education": Array [
                                                    Object {
                                                      "degree": "Bachelors of Science: Biomedical Science",
                                                      "school": "Rutgers University",
                                                    },
                                                    Object {
                                                      "degree": "Masters of Science: Biomedical Science",
                                                      "school": "Rutgers University",
                                                    },
                                                    Object {
                                                      "degree": "Docterates of Science: Biomedical Science",
                                                      "school": "Johns Hopkins University",
                                                    },
                                                  ],
                                                  "location": "Baltimore",
                                                  "name": "Alice Mendelson",
                                                  "photo": "http://lorempixel.com/160/160/",
                                                  "userId": 456,
                                                  "wikis": Array [],
                                                }
                              }
                    />
          </div>
</Connect(Page)>,
        "_debugID": 2,
        "_renderedOutput": <Connect(Page)>
          <div
                    className="profiles"
          >
                    <ProfileCard
                              handleRoute={[Function]}
                              profile={
                                        Object {
                                                  "currentProject": "Duis in nunc ultrices, viverra lorem sed, suscipit mauris. Nullam maximus, sapien non facilisis mattis, magna ex sodales ipsum, eget tristique sapien ante quis libero. Praesent pretium velit ac odio egestas fringilla. Cras nec tortor dignissim, efficitur ligula nec, imperdiet odio. Etiam in augue felis. Duis iaculis placerat vulputate. Aliquam in mauris tempor risus egestas blandit ac ut lacus.",
                                                  "education": Array [
                                                    Object {
                                                      "degree": "Bachelors of Engineering: Mechanical",
                                                      "school": "Carnegie Mellon University",
                                                    },
                                                    Object {
                                                      "degree": "Masters of Engineering: Mechanical",
                                                      "school": "Carnegie Mellon University",
                                                    },
                                                    Object {
                                                      "degree": "Docterates of Engineering: Mechanical",
                                                      "school": "Massachusetts Institute of Technology",
                                                    },
                                                  ],
                                                  "location": "Boston",
                                                  "name": "Sven Motorson",
                                                  "photo": "http://lorempixel.com/160/160/",
                                                  "userId": 123,
                                                  "wikis": Array [
                                                    Object {
                                                      "title": "Engine",
                                                      "uriParam": "engine",
                                                    },
                                                    Object {
                                                      "title": "Electric motor",
                                                      "uriParam": "Electric_motor",
                                                    },
                                                  ],
                                                }
                              }
                    />
                    <ProfileCard
                              handleRoute={[Function]}
                              profile={
                                        Object {
                                                  "currentProject": "In mattis arcu arcu, convallis volutpat eros consequat vitae. Pellentesque ultricies vulputate magna nec facilisis. Nullam consectetur risus orci, tincidunt molestie libero tempus eu. Vivamus a mauris vel nunc gravida laoreet finibus eu erat. Vestibulum sollicitudin volutpat nisl ac rhoncus. Nulla volutpat elit nec ante molestie, lacinia egestas neque molestie. Nulla nulla purus, ornare non vehicula at, blandit vel purus. Vivamus vitae ipsum id dui porttitor malesuada aliquam quis lacus. Cras quam nisi, finibus ac tellus quis, lobortis convallis nisl. Mauris viverra imperdiet placerat. Ut in eros sit amet ligula viverra luctus. Etiam ultrices ultricies pellentesque. Fusce malesuada vehicula egestas. Orci varius natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus.",
                                                  "education": Array [
                                                    Object {
                                                      "degree": "Bachelors of Science: Biomedical Science",
                                                      "school": "Rutgers University",
                                                    },
                                                    Object {
                                                      "degree": "Masters of Science: Biomedical Science",
                                                      "school": "Rutgers University",
                                                    },
                                                    Object {
                                                      "degree": "Docterates of Science: Biomedical Science",
                                                      "school": "Johns Hopkins University",
                                                    },
                                                  ],
                                                  "location": "Baltimore",
                                                  "name": "Alice Mendelson",
                                                  "photo": "http://lorempixel.com/160/160/",
                                                  "userId": 456,
                                                  "wikis": Array [],
                                                }
                              }
                    />
          </div>
</Connect(Page)>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Profiles
    profiles={
        Array [
            Object {
              "currentProject": "Duis in nunc ultrices, viverra lorem sed, suscipit mauris. Nullam maximus, sapien non facilisis mattis, magna ex sodales ipsum, eget tristique sapien ante quis libero. Praesent pretium velit ac odio egestas fringilla. Cras nec tortor dignissim, efficitur ligula nec, imperdiet odio. Etiam in augue felis. Duis iaculis placerat vulputate. Aliquam in mauris tempor risus egestas blandit ac ut lacus.",
              "education": Array [
                Object {
                  "degree": "Bachelors of Engineering: Mechanical",
                  "school": "Carnegie Mellon University",
                },
                Object {
                  "degree": "Masters of Engineering: Mechanical",
                  "school": "Carnegie Mellon University",
                },
                Object {
                  "degree": "Docterates of Engineering: Mechanical",
                  "school": "Massachusetts Institute of Technology",
                },
              ],
              "location": "Boston",
              "name": "Sven Motorson",
              "photo": "http://lorempixel.com/160/160/",
              "userId": 123,
              "wikis": Array [
                Object {
                  "title": "Engine",
                  "uriParam": "engine",
                },
                Object {
                  "title": "Electric motor",
                  "uriParam": "Electric_motor",
                },
              ],
            },
            Object {
              "currentProject": "In mattis arcu arcu, convallis volutpat eros consequat vitae. Pellentesque ultricies vulputate magna nec facilisis. Nullam consectetur risus orci, tincidunt molestie libero tempus eu. Vivamus a mauris vel nunc gravida laoreet finibus eu erat. Vestibulum sollicitudin volutpat nisl ac rhoncus. Nulla volutpat elit nec ante molestie, lacinia egestas neque molestie. Nulla nulla purus, ornare non vehicula at, blandit vel purus. Vivamus vitae ipsum id dui porttitor malesuada aliquam quis lacus. Cras quam nisi, finibus ac tellus quis, lobortis convallis nisl. Mauris viverra imperdiet placerat. Ut in eros sit amet ligula viverra luctus. Etiam ultrices ultricies pellentesque. Fusce malesuada vehicula egestas. Orci varius natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus.",
              "education": Array [
                Object {
                  "degree": "Bachelors of Science: Biomedical Science",
                  "school": "Rutgers University",
                },
                Object {
                  "degree": "Masters of Science: Biomedical Science",
                  "school": "Rutgers University",
                },
                Object {
                  "degree": "Docterates of Science: Biomedical Science",
                  "school": "Johns Hopkins University",
                },
              ],
              "location": "Baltimore",
              "name": "Alice Mendelson",
              "photo": "http://lorempixel.com/160/160/",
              "userId": 456,
              "wikis": Array [],
            },
          ]
    }
    push={[Function]}
/>,
}
`;
